src/iso.h

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright (C) 2016 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#ifndef ISO_H_
+#define ISO_H_
+
+#include "kerneltypes.h"
+
+struct iso_file {
+    int ata_unit;
+    int cur_offset;    // offset within file data
+    int extent_offset;    // offset to actual location on disk of start of file
+    int data_length;
+    char pname[256];
+};
+
+struct iso_dir {
+    int ata_unit;    // On which ATA unit this directory record exists
+    int extent_offset;  //location of the directory's extent
+    int cur_offset;  //offset to the front of the next directory record to read
+    int data_length;    // Length of data in directory record
+};
+
+
+struct directory_record {
+    uint8_t length_of_record;
+    uint8_t length_of_ext_record;
+    char loc_of_ext[8];   //Characters of hex for location of extent, first 4 are little endian, second 4 are big endian
+    char data_length[8];  //Characters of hex for the length of the directory record, first 4 are little endian, second 4 are big endian
+    uint8_t rec_date_time[7];   // timestamp of last modification
+    char file_flags[1];
+    char file_flags_interleaved[1];
+    uint8_t interleave_gap_size;
+    char vol_seq_num[4];
+    uint8_t len_identifier;
+    char file_identifier[31];  //max file id with extent is 30, +1 for null
+};
+
+/**
+ * @brief Opens the file specified
+ * @details Attempts to find and open the file specified by the pname on the given ata_unit,
+ * calls kmalloc to create return value
+ *
+ * @param pname The absolute path name to search for the file at
+ * @param ata_unit The ata unit to search for the file on
+ * @return Pointer to freshly allocated iso_file at byte 0, null if not found or on error
+ */
+struct iso_file *iso_fopen(const char *pname, int ata_unit);
+
+/**
+ * @brief Closes the file
+ * @details Calls free on the file if not null.
+ *
+ * @param file Pointer to the file to be closed
+ * @return 0 if successfully closed, -1 if not.
+ */
+int iso_fclose(struct iso_file *file);
```

Comment:
I guess what I'm asking is have we considered not mirroring the unix functions? We are making our own OS, we can do what we want to. For instance here, the only reason this would fail is that the file was NULL. Can we indicate that in the documentation?